staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -387,9 +387,17 @@ func (c *crdHandler) updateCustomResourceDefinition(oldObj, _ interface{}) {
 	glog.V(4).Infof("Updating customresourcedefinition %s", oldCRD.Name)
 
 	c.customStorageLock.Lock()
+	defer c.customStorageLock.Unlock()
+
 	storageMap := c.customStorage.Load().(crdStorageMap)
-	delete(storageMap, oldCRD.UID)
-	c.customStorageLock.Unlock()
+	storageMap2 := make(crdStorageMap)
+
+	for k, v := range storageMap {
```

Comment:
can you put a comment above like "Copy because we cannot write to storageMap without a race as it is used without locking elsewhere